spacy/tokens/_dict_proxies.py

```diff
@@ -43,6 +43,11 @@ def copy(self, doc: Optional["Doc"] = None) -> "SpanGroups":
             doc = self._ensure_doc()
         return SpanGroups(doc).from_bytes(self.to_bytes())
 
+    def setdefault(self, key, default=None):
+        if not isinstance(default, SpanGroup):
```

Comment:
What if it's an `Iterable["Span"]` which we do support for `__setitem__`? We could also call the `self._make_span_group` functionality?

Reply:
Ah, I think I was originally thinking we'd have to enforce the `name` and then there's be no way to support a list, but if we've abandoned all hope there, then we could use `_make_span_group`. (I kind of want to disable it entirely because it's less efficient that other options once you have this expensive / unused object creation here (vs. `None` or `[]`), but I guess people can make their own decisions.)